test/testdrive/float.td

```diff
@@ -0,0 +1,58 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# This test exercises decimals at the boundary (e.g., by sending them through
+# pgwire). Operations on decimals are more thoroughly tested in decimal.slt.
+
+$ set schema={
+    "type": "record",
+    "name": "envelope",
+    "fields": [
+      {
+        "name": "before",
+        "type": [
+          {
+            "name": "row",
+            "type": "record",
+            "fields": [
+              {
+                "name": "a",
+                "type": "double"
+              }
+            ]
+          },
+          "null"
+        ]
+      },
+      { "name": "after", "type": ["row", "null"] }
+    ]
+  }
+
+$ kafka-create-topic topic=data
+
+# Circumvents `i128` limit on literals parsed by MZ
+$ kafka-ingest format=avro topic=data schema=${schema} timestamp=1
+{"before": null, "after": {"row":{"a": 1.2}}}
+{"before": null, "after": {"row":{"a": 1.8e40}}}
+
+$ kafka-ingest format=avro topic=data schema=${schema} timestamp=2
+{"before": null, "after": null}
+
+> CREATE SOURCE data
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-${testdrive.seed}'
+  FORMAT AVRO USING SCHEMA '${schema}' ENVELOPE DEBEZIUM
+
+> CREATE MATERIALIZED VIEW data_view as SELECT * from data
+
+> SELECT * FROM data_view
+1.2
+18000000000000000000000000000000000000000
+
+! SELECT a::float4 FROM data_view
+float out of range
```

Comment:
Just mentioned this in Slack, but you should be able to input these floats exactly by using the string literal syntax (`'1.2'::float4`). If that does work, can we make these SLTs instead? 🤞

Reply:
Yep! Will do that in just a sec. ty for the pointer.